exercises/concept-wip/crystal-hunter/.docs/introduction.md

```diff
@@ -0,0 +1,116 @@
+# Bools
+
+Crystal has a type know as [`Bool`][bools], it is used to represent the values `true` and `false`.
+With the bool type can you use logical operators(`!`, `||`, `&&`) which can be used to combine bools and change their values.
+
+```crystal
+true && true
+# => True
+
+true && false
+# => False
+
+false && false
+# => false
+
+false || true
+# => True
+false || false
+# => false
+
+!false
+# => true
+
+!true
+# => false
+```
```

Comment:
One of the feedback that @iHiD gave me about writing concept exercise introductions is that they should limit code examples to minimum. The goal is to force students to learn the text and really understand it without giving them the option to mindlessly copy-paste everything.

Reply:
Well I am not sure if I agree with Jeremy on that one. I think forcing the student to learn by text could make learning difficult since some students will be visual learners rather than textual. And have easier-to-follow code rather than reading text. With that said I think having text also is valuable. If there is a location where I could be more detailed, so let me know. For the earlier concepts I have written (not for crystal), I got praised by other maintainers on exercism for including a lot of examples. Thereby I have a bit mixed feelings about this.

Also another problem is that crystals documentation is not that good and is something the devs are aware of. If it is covered in the "doc" section, then is the doc good/okay but if it isn't it is hard to find good references.
I have gotten quite used to reading the source code of the language, but I do not think a lot of developers read the source code to understand how to do stuff.

I get the point, of reducing examples, they take place and some students like very consisted content so it is hard to make a concept that suits everybody.

It is again something I will have to think over, the snippet you mentioned has been reduced.

Reply:
I think the point Angelika is making here is that by providing all the pairs written out, there's a risk the exercise becomes utterly trivial (basically people can just copy/paste) things. And then someone doesn't actually learn anything. It's important that someone can struggle a little to solve the exercises in order to actually learn.

> since some students will be visual learners rather than textual

Just on this note, there's pretty strong consensus among experts that different learning styles are a myth, which is something I was interested to learn and has changed my approach to creating content (e.g. https://fee.org/articles/learning-styles-don-t-actually-exist-studies-show/). 

I think probably **everyone** learns better with visuals and I am totally pro having code and visuals that demonstrate something. However, we have to be careful not just to give answers away and to create an appropriate amount of struggle to provide break-thought mental model moments.

---

Note: I'm not commenting on this specific example (I've not read the PR) but on the general principle.

Reply:
I think it is a valuable point both of you are bringing. I agree that the concepts should not be written in a way to allow "copy-pasting". Although as I tried to point out if you can't actually figure out what to do, could lead to frustrations. As I pointed out earlier crystal doesn't have that deeply written documentation when approaching difficult concepts. But if the student doesn't fully learn a concept, so will that likely lead to frustrations later. It is a bit of a balance to make the student think enough for themself, but not too much to make it frustrating. Thereby is it important to reveal just the right amount of information or as I would say in Swedish "lagom".

If there is a part where you think I am too "revling" or give too little information for that side too, please let me know and I will look over it/ you can suggest an improvement.

That with learning styles could be discussable, the article you sent is a debate article, and I agree that everyone could probably learn equally with every method. That doesn't mean that people don't have a preference for different learning styles. I dislike learning programming through youtube videos, although there are programming tutorial videos with 40 million views so somebody has to watch them, I rather have interactive learning. Meanwhile, when I learned chemistry I loved learning through videos and hated laborations (I was afraid to get some dangerous substance on my body). Although that doesn't mean that other students didn't appreciate those.